### PR TITLE
Warning overhaul

### PR DESCRIPTION
**Pull request by ZehMatt**
Based on #1854 so I'll have to rebase if thats in. This removes a couple disabled warnings and added `[[maybe_unused]]` for the time being, ordinarily unused parameters should be nameless according to https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f9-unused-parameters-should-be-unnamed but the signatures of the functions or use may change so I sticked to `[[maybe_unused]]` for most of it, once Interop is gone we can adjust the function signatures to improve this.

### CODE REVIEW

File: src/OpenLoco/src/Interop/Hooks.cpp [hunk 1]
```diff
@@ -805,7 +805,7 @@ void OpenLoco::Interop::registerHooks()
 
     registerHook(
         0x004392BD,
-        [](registers& regs) FORCE_ALIGN_ARG_POINTER -> uint8_t {
+        []([[maybe_unused]] registers& regs) FORCE_ALIGN_ARG_POINTER -> uint8_t {
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Interop/Hooks.cpp:808]: can't we just have these unnamed the hooks aren't going to change apart from be deleted.

File: src/OpenLoco/src/Map/MapGenerator.cpp [hunk 2]
```diff
@@ -46,7 +46,7 @@ namespace OpenLoco::Map::MapGenerator
         int32_t const height;
         int32_t const pitch;
 
-        HeightMap(int32_t width, int32_t height, int32_t pitch, uint8_t baseHeight)
+        HeightMap(int32_t width, int32_t height, int32_t pitch, [[maybe_unused]] uint8_t baseHeight)
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Map/MapGenerator.cpp:49]: this should just be deleted.

File: src/OpenLoco/src/Interop/Hooks.cpp (hunk 1 above)
Comment (ZehMatt) [reply src/OpenLoco/src/Interop/Hooks.cpp:808]: True, i'll change it.

File: src/OpenLoco/src/Windows/CompanyList.cpp [hunk 3]
```diff
@@ -195,6 +195,7 @@ namespace OpenLoco::Ui::Windows::CompanyList
         }
 
         // 0x00437BE1
+        // FIXME: rhs is not used.
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Windows/CompanyList.cpp:198]: has now been addressed

File: cmake/OpenLocoUtility.cmake [hunk 4]
```diff
@@ -58,14 +58,8 @@ function(loco_target_compile_link_flags TARGET)
         /W4                      # Warning level 4
                                  # Poke holes in W4 due to our interop code
         /wd4068                  #   4068: unknown pragma
-        /wd4091                  #   4091: 'keyword' : ignored on left of 'type' when no variable is declared
-        /wd4100                  #   4100: 'identifier' : unreferenced formal parameter
```
Comment (duncanspumpkin) [inline cmake/OpenLocoUtility.cmake:62]: isn't -Wno-unused-parameter the same as this. Could you remove it for gcc/clang. L72.

Comment (ZehMatt) [reply cmake/OpenLocoUtility.cmake:62]: I think so, I'll give it a try.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**